app/dojo.py

```diff
@@ -0,0 +1,141 @@
+
+	def add_new_member(self, person_category, person_name, wants_accomodation="N"):
+		"""Add a new member to the Dojo"""
+
+		self.person_name = person_name
+		self.person_category = person_category
+		self.wants_accomodation = wants_accomodation
+
+		"""Check if the person already exists"""
+		if self.person_name in self.fellows or self.person_name in self.staff:
+
+			print("There is already a person with with the name " + self.person_name + ". Please try again")
+			raise ValueError
+
+		
+		elif self.person_category not in ["Staff", "Fellow"]:
+			"""Check if the category is valid"""
+
+			print("The Dojo doesn't have the membership you've entred")
+
+		
+		elif self.wants_accomodation == "Y":
+		#Allocate accomodation to Fellows only and offices to Fellows and Staff
+			if self.person_category == "Fellow":
+				
+				new_fellow = Fellow(self.person_name)
+				allocated_office = choice([key for key, value in self.offices.items()])
+				allocated_living_space = choice([key for key, value in self.living_spaces.items()])
+				self.fellows.append(str(new_fellow))
+				self.offices[allocated_office]['Fellows'].append(str(new_fellow))
+				self.living_spaces[allocated_living_space].append(str(new_fellow))
+				print("Fellow " + self.person_name + " has been successfully added!")
+				print(self.person_name + " has been allocated the " + allocated_office + " office")
+				print(self.person_name + " has been allocated the " + allocated_living_space + " living space")
+				
+
+			else:
+
+				new_staff = Staff(self.person_name)
+				allocated_office = choice([key for key, value in self.offices.items()])
+				self.staff.append(str(new_staff))
+				self.offices[allocated_office]['Staff'].append(str(new_staff))
+				print("Staff " + self.person_name + " has been successfully added!")
+				print(self.person_name + " has been allocated the " + allocated_office + " office")
+				print("Living Spaces are only available for fellows. No room was allocated to " + self.person_name)
+				
+
+		else:
+
+		#allocate offices to Staff and to Fellows who don't want accomodation"""
+			if self.person_category == "Fellow":
+				
+				new_fellow = Fellow(self.person_name)
+				allocated_office = choice([key for key, value in self.offices.items()])
+				self.fellows.append(str(new_fellow))
+				self.offices[allocated_office]['Fellows'].append(str(new_fellow))
+				print("Fellow " + self.person_name + " has been successfully added!")
+				print(self.person_name + " has been allocated the " + allocated_office + " office")
+				
+			else:
+
+				new_staff = Staff(self.person_name)
+				allocated_office = choice([key for key, value in self.offices.items()])
+				self.staff.append(str(new_staff))
+				self.offices[allocated_office]['Staff'].append(str(new_staff))
+				print("Staff " + self.person_name + " has been successfully added!")
+				print(self.person_name + " has been allocated the " + allocated_office + " office")
+								
+
```

Comment:
There is a lot of mixup in add_person method. Consider the problem definition. You can allocate office then check if the person is fellow and wants accommodation allocate livingspace.